hazelcast/src/main/java/com/hazelcast/collection/impl/collection/CollectionContainer.java

```diff
@@ -162,6 +162,15 @@ public void addAllBackup(Map<Long, Data> valueMap) {
         getMap().putAll(map);
     }
 
+    public void replaceBackups(Collection<CollectionItem> backupItems) {
+        getMap().clear();
+        Map<Long, CollectionItem> map = createHashMap(backupItems.size());
+        for (CollectionItem backupItem : backupItems) {
+            map.put(backupItem.getItemId(), backupItem);
+        }
+        getMap().putAll(map);
```

Comment:
Can we skip this additional map and add to the backup map directly?

Reply:
Good one. I also fixed a possible NPE and moved this the logic to the operation itself.